src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -87,6 +81,106 @@ public async Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query)
             }
         }
 
+        /// <summary>
+        /// Use this method to receive large quantities of data from Kusto in a "stream". 
+        /// This method assumes the query is returning a single schema result set. 
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public IEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery query)
+        {
+            var client = GetProvider();
+            var properties = BuildClientRequestProperties(query);
+            properties.SetOption(ClientRequestProperties.OptionResultsProgressiveEnabled, true);
+
+            string text = BuildQueryText(query);
+
+            ProgressiveDataSet pDataSet = client.ExecuteQueryV2Async(
+                DatabaseName,
+                text,
+                properties).GetAwaiter().GetResult();
+
+            int tableCompletionCount = 0;
+
+            using (var frames = pDataSet.GetFrames())
+            {
+                while (frames.MoveNext())
+                {
+                    var frame = frames.Current;
+
+                    switch (frame.FrameType)
+                    {
+                        case FrameType.TableFragment:
+                            {
+                                var content = frame as ProgressiveDataSetDataTableFragmentFrame;
+                                object[] row;
+                                while(GetNextRow(content, out row))
```

Comment:
```suggestion
                                while(GetNextRow(content, out object[] row))
```
Related to my other comment, this makes it a touch clearer it's not some existing, reused object. And less lines of code!